### PR TITLE
Filter Repeating Logs

### PR DESCRIPTION
**Pull request by NihalHarish**
### Description of changes:
- Filter out logs that repeat more than 5 times to prevent log spam

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by leleamol — COMMENTED** [review]
The fix looks good. In addition, can we reduce the logging level to debug for 
logger.info(
                    "Checkpoints not updated. There are no checkpoint files created yet, to be updated"
                )

**Comment by NihalHarish** [comment]
> Checkpoints not updated. There are no checkpoint files created yet, to be updated

Do I change the level for all the logs in the file or just the one you have mentioned?

**Review by leleamol — APPROVED** [review]

File: smdebug/core/logger.py [hunk 1]
```diff
@@ -19,6 +20,18 @@ def filter(self, record):
         return record.levelno < self.level
 
 
+class DuplicateLogFilter:
+    """Filters duplicate messages to prevent spamming users"""
+
+    def __init__(self):
+        self.msgs = defaultdict(int)
+        self.repeat_threshold = 5
```
Comment (ndodda-amazon) [inline smdebug/core/logger.py:28]: Can we lower this to 2 or 3?

Also, can we make this a constant?

**Review by ndodda-amazon — APPROVED** [review]
LGTM! Left a minor comment.

**Pull request merged**